pkg/xds/secrets/secrets.go

```diff
@@ -75,11 +75,20 @@ func NewSecrets(caProvider CaProvider, identityProvider IdentityProvider, metric
 		return nil, err
 	}
 
+	otherMeshesCertGenerationsMetric := prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
feels like this metric is only used for assertions in tests, what we're actually trying to test is `shouldGenerateCerts` - should we just restructure the struct? I know it's already been written that way but it seems kind of clunky to add a metric to cover a test case 🤷